test/NuGet.Core.Tests/NuGet.Packaging.Test/ReferencesInNuspecMatchRefAssetsRuleTests.cs

```diff
@@ -0,0 +1,377 @@
+                "- Add MyHelpers.dll to the net462 reference group in the nuspec" + Environment.NewLine;
+            Assert.Equal(singleIssue.Message, expectedMessage);
+        }
+
+        [Fact]
+        public void GenerateWarnings_PackageWithAssetsMissingFromTheRefFolder_ShouldWarn()
+        {
+            //Arrange & Act
+            var rule = new ReferencesInNuspecMatchRefAssetsRule();
+            var missingItems = new List<MissingReference>
+            {
+                new MissingReference("ref", "net462", new string[] { "MyHelpers.dll"})
+            };
+            var issues = rule.GenerateWarnings(missingItems);
+
+            //Assert
+            Assert.Equal(issues.Count(), 1);
+            var singleIssue = issues.Single(t => t.Code == NuGetLogCode.NU5131);
+            var expectedMessage = "References were found in the nuspec, but some reference assemblies were not found in both the nuspec and ref folder. Add the following reference assemblies:" + Environment.NewLine +
+                "- Add MyHelpers.dll to the ref/net462/ directory" + Environment.NewLine;
+            Assert.Equal(singleIssue.Message, expectedMessage);
+        }
+
+        [Fact]
+        public void GenerateWarnings_PackageWithSubfoldersThatAreMismatched_ShouldWarn()
+        {
+            //Arrange & Act
+            var rule = new ReferencesInNuspecMatchRefAssetsRule();
+            var missingItems = new List<MissingReference>
+            {
+                new MissingReference("ref", "net472", new string[] { "MyLib.dll"}),
+                new MissingReference("nuspec", "net462", new string[] { "MyHelpers.dll"})
+            };
+            var issues = rule.GenerateWarnings(missingItems);
+
+            //Assert
+            Assert.Equal(1 ,issues.Count());
+            var singleIssue = issues.Single(t => t.Code == NuGetLogCode.NU5131);
+            var expectedMessage = "References were found in the nuspec, but some reference assemblies were not found in both the nuspec and ref folder. Add the following reference assemblies:" + Environment.NewLine +
+                "- Add MyLib.dll to the ref/net472/ directory" + Environment.NewLine +
+                "- Add MyHelpers.dll to the net462 reference group in the nuspec" + Environment.NewLine;
+            Assert.Equal(singleIssue.Message, expectedMessage);
+        }
+
+        [Fact]
+        public void GenerateWarnings_PackageWithSubFoldersThatAreCorrect_ShouldNotWarn()
+        {
+            //Arrange & Act
+            var rule = new ReferencesInNuspecMatchRefAssetsRule();
+            var missingItems = Array.Empty<MissingReference>();
+            var issues = rule.GenerateWarnings(missingItems);
+
+            //Assert
+            Assert.Empty(issues);
+        }
+
+        [Fact]
+        public void Compare_NuspecHasFilesWithNoSpecificTfmWithMatchingRefFiles_ShouldBeEmpty()
```

Comment:
move this Compare test with the other Compare tests, so the Compare and GenerateWarnings tests are "grouped" together.

I don't know if you agree, but I think the GenerateWarnings tests above are easier to visually verify. If you could do the same with the GenerateWarnings tests below this Compare test, I think that would be good.

Reply:
I do want to have some tests for the warning message.